libsrc/cx16/tgi/cx640p1.s

```diff
@@ -0,0 +1,677 @@
+        lda     #$01
+        sta     VERA::ADDR+2    ; write color RAM @ $1FA00
+
+        lda     palette
+        asl
+        tay
+        lda     veracolors,y
+        sta     VERA::DATA0
+
+        inc     VERA::ADDR      ; $1FA01
+
+        lda     palette
+        asl
+        tay
+        iny     ; second byte of color
+        lda     veracolors,y
+        sta     VERA::DATA0
+
+        ; set foreground color from palette color 1
+        inc     VERA::ADDR      ; $1FA02
+
+        lda     palette+1
+        asl
+        tay
+        lda     veracolors,y
+        sta     VERA::DATA0
+
+        inc     VERA::ADDR      ; $1FA03
+
+        lda     palette+1
+        asl
+        tay
+        iny     ; second byte of color
+        lda     veracolors,y
+        sta     VERA::DATA0
+        rts
+
+; ------------------------------------------------------------------------
+; SETCOLOR: Set the drawing color (in .A). The new color already is checked
+; to be in a valid range (0..maxcolor).
+;
+; Must set an error code: NO (will be called only if color OK)
+
+SETCOLOR:
+        tax
+        beq     @L1
+        lda     #$FF
+@L1:    sta     BITMASK
+        stx     color
+        rts
+
+; ------------------------------------------------------------------------
+; GETPALETTE: Return the current palette in .XA. Even drivers that cannot
+; set the palette should return the default palette here, so there's no
+; way for this function to fail.
+;
+; Must set an error code: NO
+
+GETPALETTE:
+        lda     #<palette
+        ldx     #>palette
+        rts
+
+; ------------------------------------------------------------------------
+; GETDEFPALETTE: Return the default palette for the driver in .XA. All
+; drivers should return something reasonable here, even drivers that don't
+; support palettes, otherwise the caller has no way to determine the colors
+; of the (not changable) palette.
+;
+; Must set an error code: NO (all drivers must have a default palette)
+
+GETDEFPALETTE:
+        lda     #<defpalette
+        ldx     #>defpalette
+        rts
+
+; ------------------------------------------------------------------------
+; SETPIXEL: Draw one pixel at X1/Y1 = ptr1/ptr2 with the current drawing
+; color. The co-ordinates passed to this function never are outside the
+; visible screen area, so there is no need for clipping inside this function.
+;
+; Must set an error code: NO
+
+SETPIXEL:
+        jsr CALC
+
+        stx TEMP
+
+        lda ADDR
+        ldy ADDR+1
+        ldx #$00
+
+        sta VERA::ADDR
+        sty VERA::ADDR + 1
+        stx VERA::ADDR + 2
+
+        ldx TEMP
+
+        lda BITMASK
+        beq @ahead
+
+        ; if BITMASK = $00, white is line color
+        ; Set the bit in the byte at VERA_DATA0
+        lda VERA::DATA0      ; Load the byte at memory address
+        ora bitMasks1,X      ; OR with the bit mask
+        sta VERA::DATA0      ; Store back the modified byte
+        rts
+
+    @ahead:
```

Comment:
Please make all labels start at the beginning of a line

Reply:
Got it, thank you